### DISCUSSION TITLE
[Q] How can a call to Insert rows into BigQuery using the client in c++ be made?

### DISCUSSION THREAD

**Discussion by yagnasrinath** [googleapis/google-cloud-cpp#6809] [Q&A] [answered]
I see that old client listed here http://google.github.io/google-api-cpp-client/latest/available_service_apis.html seemed to have all the APIs including inserts. But the issue with this one is that there is no documentation anywhere that allows one to build successfully. Also, most of the dependencies client library used by the generated code that is part of this library are deprecated according to their github READMEs. Also, the json library used by most of these libs seems to be very old and the wrappers to json library used by these libs from the link seems to be missing from the libraries.

Instead of relying on the library from the link, I have also tried generating library using the discovery document. Even that seems to have similar issues as mentioned above. 

Most of the docs point to this monorepo, which seems to have bigquery client. The included client doesn't seem to have this functionality. If it is present, there is no documentation related to it.

Could you please help me identify the right library and instructions to get the client built to insert rows into bigquery tables.

**Comment by coryan** [comment, marked as answer]
**TL;DR;** there is no recommended or GA library for BigQuery C++ at this point in time.

We are working to support more and more Cloud Services in `google-cloud-cpp` and these questions are useful to guide our prioritization, thanks!

**Comment by yagnasrinath** [comment]
Thank you for the response. But is there any document or help I can get to resolve the issues with the deprecated bigquery library mentioned above until it is available as part of this library?

**Comment by coryan** [comment]
>  But is there any document or help I can get to resolve the issues with the deprecated bigquery library mentioned above until it is available as part of this library?

Sadly, I do not know of any.  I have zero experience with `google-api-cpp-client`, and nobody in my team does either.

You may be able to use a REST library to send simple requests.  You probably can use the classes in `google-cloud-cpp/google/cloud/storage/oauth2` to create the authentication tokens.  I know this is not a great answer, but it is the best I can offer at this time.
